check.go

```diff
@@ -52,3 +57,27 @@ func (c *Conn) CheckDomain(domains ...string) (*DomainCheck, error) {
 // DomainCheck exists for backwards compatibility.
 // FIXME: remove/improve this.
 type DomainCheck domainCheckData
+
+func decodeCheckDomainResponse(d *Decoder) (*domainCheckData, error) {
+	d.Reset()
+	data := &domainCheckData{}
+	for {
+		t, err := d.Token()
+		if err != nil && err != io.EOF {
+			return nil, err
+		}
+		if t == nil {
+			break
+		}
+		switch node := t.(type) {
+		case xml.StartElement:
+			fmt.Printf("StartElement: %s %s\n", node.Name.Space, node.Name.Local)
+		case xml.EndElement:
+			fmt.Printf("EndElement: %s %s\n", node.Name.Space, node.Name.Local)
+		case xml.CharData:
+			fmt.Printf("CharData: %s\n", string(node))
+		}
+		fmt.Printf("Stack: %+v\n", d.Stack)
```

Comment:
Confirmed that running `TestDecodeCheckDomainResponse` produces lots of debugging output.